x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning.tsx

```diff
@@ -34,40 +40,195 @@ interface Props {
+            }
+          )}
+        </EuiCallOut>
+      )}
+      {unsearchedUnconfirmedFields && (
+        <EuiCallOut
+          color="warning"
+          iconType="alert"
+          title={i18n.translate(
+            'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.uncofirmedFieldsBannerLabel',
+            {
+              defaultMessage: 'Recently added fields are not being searched by default',
+            }
+          )}
+          data-test-subj="RelevanceTuningUnsearchedFieldsCallout"
+        >
+          <FormattedMessage
+            id="xpack.enterpriseSearch.appSearch.engine.relevanceTuning.uncofirmedFieldsErrorMessage"
+            defaultMessage="If these new fields should be searchable, turn them on here by toggling Text Search. Otherwise, confirm your new {schemaLink} to dismiss this alert."
+            values={{
+              schemaLink: (
+                <EuiLinkTo to={generateEnginePath(ENGINE_SCHEMA_PATH)}>
+                  {i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.schemaFieldsLinkLabel',
+                    {
+                      defaultMessage: 'schema fields',
+                    }
+                  )}
+                </EuiLinkTo>
+              ),
+            }}
+          />
+        </EuiCallOut>
+      )}
+      {schemaFieldsWithConflictsCount > 0 && (
+        <EuiCallOut
+          color="warning"
+          iconType="alert"
+          title={i18n.translate(
+            'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.schemaConflictsBannerLabel',
+            {
+              defaultMessage: 'Disabled fields',
+            }
+          )}
+          data-test-subj="SchemaConflictsCallout"
+        >
+          <FormattedMessage
+            id="xpack.enterpriseSearch.appSearch.engine.relevanceTuning.schemaConflictsErrorMessage"
+            defaultMessage="{schemaFieldsWithConflictsCount, number} inactive {schemaFieldsWithConflictsCount, plural, one {field} other {fields}} due to field-type conflicts. {link}"
+            values={{
+              schemaFieldsWithConflictsCount,
+              link: (
+                <EuiLink href={`${DOCS_PREFIX}/meta-engines-guide.html`} target="_blank">
+                  {i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.whatsThisLinkLabel',
+                    {
+                      defaultMessage: "What's this?",
+                    }
+                  )}
+                </EuiLink>
+              ),
+            }}
+          />
+        </EuiCallOut>
+      )}
+      <EuiSpacer />
+      {engineHasSchemaFields ? (
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <RelevanceTuningForm />
+          </EuiFlexItem>
+          <EuiFlexItem />
+        </EuiFlexGroup>
+      ) : (
+        <EuiEmptyPrompt
```

Comment:
[Not a blocker] I'm getting a flash of this EuiEmptyPrompt on page load:

![flash](https://user-images.githubusercontent.com/549407/109218757-aa760780-776b-11eb-9028-ce033bc9b389.gif)

I think adding a `dataLoading` prop to this view would ensure a smoother UI/transition. That being said, I don't think this is a blocker for this specific PR, so if you prefer we can do it in a follow-up PR (I'm happy to help out with this if you like).

If you do decide to take this on now, I'd also strongly prefer to see the EuiEmptyPrompt being an early return before line 59, and not a ternary here if possible (correct me if I'm wrong but conflicts/callouts won't show if there are no fields, so no point rendering the callouts etc.)

Reply:
For the early return, I * could* do that, but it's a little messy because I'd need to repeat all of the wrapper code. I could split it into a separate function though and have an early return.

Reply:
What do you think of something like this? Just a quick way to break it up and make the structure more readable and obvious. Instead of functions, those could be separate components, either inline or in their own files. 343d0061657cf9d5a150e404b184d37396c9d8a7

Reply:
That boils the main code down to:
```jsx
return (
    <>
      <UnsavedChangesPrompt hasUnsavedChanges={unsavedChanges} />
      <SetPageChrome trail={[...engineBreadcrumb, RELEVANCE_TUNING_TITLE]} />
      {pageHeader()}

      <FlashMessages />
      {invalidBoosts && invalidBoostsCallout()}
      {unsearchedUnconfirmedFields && unsearchedUnconfirmedFieldsCallout()}
      {schemaFieldsWithConflictsCount > 0 && schemaFieldsWithConflictsCallout()}
      <EuiSpacer />
      {engineHasSchemaFields ? (
        <EuiFlexGroup>
          <EuiFlexItem>
            <RelevanceTuningForm />
          </EuiFlexItem>
          <EuiFlexItem />
        </EuiFlexGroup>
      ) : (
        emptyCallout()
      )}
    </>
  );
```

Reply:
For the early return, does this help? 351fd33b5a1475e39e4f1544bbc5b2fe632776a5

Reply:
Loading state: 335eb1a5956077b35e0cefbac04aca2ec954b324. We had it, I just wasn't using it.

Reply:
Love it!